darts/timeseries.py

```diff
@@ -603,63 +633,55 @@ def from_dataframe(
-                    # Try to convert to integers if needed
-                    try:
-                        time_col_vals = time_col_vals.astype(int)
-                    except ValueError:
-                        pass
-
-                if np.issubdtype(time_col_vals.dtype, np.integer):
-                    # We have to check all integers appear only once to have a valid index
-                    raise_if(
-                        time_col_vals.duplicated().any(),
-                        "The provided integer time index column contains duplicate values.",
-                    )
+            if time_col not in df.columns:
+                raise_log(AttributeError(f"time_col='{time_col}' is not present."))
 
-                    start_idx, stop_idx = min(time_col_vals), max(time_col_vals) + 1
+            time_index = pd.Index([])
+            time_col_vals = df[time_col]
 
-                    # All the integers in the range have to be present
-                    raise_if_not(
-                        stop_idx - start_idx == len(df),
-                        "The provided integer time index column does not contain all integers in the range.",
-                    )
+            if np.issubdtype(time_col_vals.dtype, object):
+                # Try to convert to integers if needed
+                try:
+                    time_col_vals = time_col_vals.astype(int)
+                except ValueError:
+                    pass
 
-                    # Temporarily use an Int64Index (soon to be NumericIndex) to sort the values,
-                    # then replace by a RangeIndex.
-                    series_df.index = time_col_vals
-                    series_df = series_df.sort_index()
-                    time_index = pd.RangeIndex(
-                        start=start_idx, stop=stop_idx, step=1, name=time_col
-                    )
-                elif np.issubdtype(time_col_vals.dtype, object):
-                    # The integer conversion failed; try datetimes
-                    try:
-                        time_index = pd.DatetimeIndex(time_col_vals)
-                    except ValueError:
-                        raise_log(
-                            AttributeError(
-                                "'time_col' is of 'object' dtype but doesn't contain valid timestamps"
-                            )
-                        )
-                elif np.issubdtype(time_col_vals.dtype, np.datetime64):
+            if np.issubdtype(time_col_vals.dtype, np.integer):
+                # We have to check all integers appear only once to have a valid index
+                raise_if(
+                    time_col_vals.duplicated().any(),
+                    "The provided integer time index column contains duplicate values.",
+                )
+
+                # Temporarily use an Int64Index (soon to be NumericIndex) to sort the values,
+                # and replace by a RangeIndex in `TimeSeries.from_xarray()`
```

Comment:
Are these comments still relevant? I have the impression that this is not using Int64Index (which are going to be deprecated by pandas soon).